dali/base/dafdesc.cpp

```diff
@@ -3581,20 +3610,75 @@ bool getDefaultSpillPlane(StringBuffer &ret)
 
 //---------------------------------------------------------------------------------------------------------------------
 
+class CStoragePlaneAlias : public CInterfaceOf<IStoragePlaneAlias>
+{
+public:
+    CStoragePlaneAlias(IPropertyTree *_xml) : xml(_xml)
+    {
+        Owned<IPropertyTreeIterator> modeIter = xml->getElements("mode");
+        ForEach(*modeIter)
+        {
+            const char *modeStr = modeIter->query().queryProp(nullptr);
+            modes |= getAccessModeFromString(modeStr);
+        }
+    }
+    virtual AccessMode queryModes() const override { return modes; }
+    virtual const char *queryPrefix() const override { return xml->queryProp("@prefix"); }
+
+private:
+    Linked<IPropertyTree> xml;
+    AccessMode modes = AccessMode::none;
+};
+
 class CStoragePlaneInfo : public CInterfaceOf<IStoragePlane>
 {
 public:
-    CStoragePlaneInfo(IPropertyTree * _xml) : xml(_xml) {}
+    CStoragePlaneInfo(IPropertyTree * _xml) : xml(_xml)
+    {
+        Owned<IPropertyTreeIterator> srcAliases = xml->getElements("aliases");
+        ForEach(*srcAliases)
+            aliases.push_back(new CStoragePlaneAlias(&srcAliases->query()));
+    }
+    ~CStoragePlaneInfo()
+    {
+        for (auto & alias : aliases)
+            alias->Release();
+    }
 
     virtual const char * queryPrefix() const override { return xml->queryProp("@prefix"); }
     virtual unsigned numDevices() const override { return xml->getPropInt("@numDevices", 1); }
     virtual const char * queryHosts() const override { return xml->queryProp("@hosts"); }
     virtual const char * querySingleHost() const override { return xml->queryProp("@host"); }   // MORE: Likely to be changed to resolve hosts
     virtual unsigned numDefaultSprayParts() const override { return xml->getPropInt("@defaultSprayParts", 1); }
     virtual bool queryDirPerPart() const override { return xml->getPropBool("@subDirPerFilePart", isContainerized()); } // default to dir. per part in containerized mode
-
+    virtual IStoragePlaneAlias *getAliasMatch(AccessMode desiredModes) const override
+    {
+        if (AccessMode::none == desiredModes)
+            return nullptr;
+        // go through and return one with most mode matches (should there be any other weighting?)
+        unsigned bestScore = 0;
+        IStoragePlaneAlias *bestMatch = nullptr;
+        for (const auto & alias : aliases)
+        {
+            AccessMode aliasModes = alias->queryModes();
+            unsigned match = static_cast<unsigned>(aliasModes & desiredModes);
+            unsigned score = 0;
+            while (match)
+            {
+                score += match & 1;
+                match >>= 1;
+            }
+            if (score > bestScore)
+            {
+                bestScore = score;
+                bestMatch = alias;
+            }
+        }
+        return LINK(bestMatch);
+    }
 private:
     Linked<IPropertyTree> xml;
+    std::vector<IStoragePlaneAlias *> aliases;
```

Comment:
std::vector<Owned<IStoragePlaneAlias>> to avoid the destructor?

Reply:
yes, I consciously chose not to for some reason (thinking of efficiency), but not worth thinking about here, and yes I'd prefer if Owned, I'll change.